### ISSUE TITLE
Directions interface does not use exact location when selected by right-clicking

### ISSUE THREAD

**Issue by werenike** [openstreetmap/openstreetmap-website#1911]
#### Issue description
The directions interface does not use the location that I click. Instead it somehow snaps to a nearest 'node' or grid of some kind.

#### Steps to reproduce the issue
1. On the main map page, right-click on a location and select 'Directions from here'.
2. Right-click somewhere else and select 'Directions to here'.

#### What's the expected result?
I should get directions from the exact location I clicked first, to the exact location I clicked second.

#### What's the actual result?
I get directions from some point near the location I clicked first, to some point near the location I clicked second.

#### Additional details
- Directions between arbitrary points are not impossible for the engine, as is evidenced by manually clicking and dragging the start and end nodes of the generated directions to arbitrary locations (they do not snap back then).
- Possibly the cause of this behaviour is [here](https://github.com/openstreetmap/openstreetmap-website/blob/999b9c69e1af252610a2ad338d33d16273fcc22f/app/assets/javascripts/index/contextmenu.js#L22)? (I wanted to try it out but I couldn't get the dev env running)

**Comment by tomhughes** [comment]
I believe all endpoints are passed through the geocoder - fixing that will not be trivial.

**Comment by tomhughes** [comment]
The line you pointed to won't have any (significant effect) unless you really want centimetric precision.

**Comment by tomhughes** [comment]
Yes https://github.com/openstreetmap/openstreetmap-website/blob/999b9c69e1af252610a2ad338d33d16273fcc22f/app/assets/javascripts/index/directions.js#L69 is the real problem because it passes the value through the geocoder which likely snaps to the nearest place it knows about.

**Comment by MareinK** [comment]
Thanks for your comments. I'm not familiar with the codebase - would it be an option to not use the geocoder in that instance?

**Comment by krubokrubo** [comment]
+1. This is a real problem in remote places where the map is sparse.  I can see the road that I know contains my destination, but I can't select my destination on the road.  I can only select "the closest node" which may be far away.

**Comment by BjornRasmussen** [comment]
This is a problem for me every time I am trying to test connectivity.  😕

**Comment by AntonKhorev** [comment]
What's the problem here? Do you want some lines connecting the endpoint markers to the actual route?

**Comment by BjornRasmussen** [comment]
The chosen start and end point don't have to have a line connecting to the start of the navigation.  The issue is that the chosen point jumps to the nearest address, poi, etc after you select it most of the time.

https://github.com/user-attachments/assets/485571e5-4b4a-4631-b3bd-d0e493593f1b

**Comment by AntonKhorev** [comment]
If the problem is that the markers jump, it may be fixed in #5064.
